src/components/scenegraph/Toolbar.js

```diff
@@ -69,13 +141,35 @@ export default class Toolbar extends React.Component {
    * Try to write changes with aframe-inspector-watcher.
    */
   writeChanges = () => {
-    const xhr = new XMLHttpRequest();
-    xhr.open('POST', 'http://localhost:51234/save');
-    xhr.onerror = () => {
-      alert('aframe-watcher not running. This feature requires a companion service running locally. npm install aframe-watcher to save changes back to file. Read more at supermedium.com/aframe-watcher');
-    };
-    xhr.setRequestHeader('Content-Type', 'application/json');
-    xhr.send(JSON.stringify(AFRAME.INSPECTOR.history.updates));
+    const baseUrl = process.env.REACT_APP_ADMIN_BACKEND_URL;
+    const apiEndpointScene = AFRAME.scenes[0].getAttribute("id").replace("-scene", "");
+    const baseEndpoint = process.env.REACT_APP_ADMIN_BASE_ENDPOINT;
+    const getUrl = baseUrl + baseEndpoint + "scene/" + apiEndpointScene;
+    let objects = this.state.objects;
+    let objectChanges = [];
+    for(var id in AFRAME.INSPECTOR.history.updates)
+      objectChanges.push([parseInt(id.replace("-obj", "")), AFRAME.INSPECTOR.history.updates[id]]);
+    objectChanges.sort();
+    let i = 0;
+    let j = 0;
+    while(i < objects.length && j < objectChanges.length){
+      if (objects[i].id == objectChanges[j][0]){
+        let curChanges = objectChanges[j][1];
+        for (const prop in curChanges){
+          if (prop == "position" || prop == "scale" || prop == "rotation"){
+            objects[i][prop] = curChanges[prop].split(" ").map(Number);
+          } else if (prop == "gltf-model"){
+            objects[i]["asset_id"] = this.state.linkToIdMap[curChanges[prop]];
+          }
```

Comment:
Is it possible for prop to be none of them? Like none of `position, scale, rotation, gltf-model`? If it is possible, what are the other changes possible?

Reply:
So far haven't ran into others, I think it might be possible through the add components option in the aframe-inspector